dev/archery/archery/utils/lint.py

```diff
@@ -230,6 +230,7 @@ def python_linter(src, fix=False):
 
     yield LintResult.from_cmd(
         flake8("--extend-exclude=" + ','.join(flake8_exclude),
+               "--config=python/setup.cfg",
                setup_py, src.pyarrow, os.path.join(src.python, "examples"),
```

Comment:
For this to work in the docker lint job we need to specify the absolute path as archery is called from outside of the arrow. (see [here](https://github.com/apache/arrow/actions/runs/3757323229/jobs/6384396332#step:5:837))

This should™ do it:
```suggestion
        flake8("--extend-exclude=" + ','.join(flake8_exclude),
               "--config=" + os.path.join(src.python, "setup.cfg"),
               setup_py, src.pyarrow, os.path.join(src.python, "examples"),
```